managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1906,7 +1906,9 @@ public void asyncMarkDelete(final Position position, Map<String, Long> propertie
 
         lock.writeLock().lock();
         try {
-            newPosition = setAcknowledgedPosition(newPosition);
+            if (!newPosition.equals(markDeletePosition)) {
+                newPosition = setAcknowledgedPosition(newPosition);
```

Comment:
I don't understand why we are skipping this method, because I found that this judgment already exists in `setAcknowledgedPosition`.